src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -1435,19 +1435,22 @@ private CSharpSyntaxNode GetBindingRoot(CSharpSyntaxNode node)
             }
 #endif
 
-            for (CSharpSyntaxNode current = node; current != this.Root; current = current.ParentOrStructuredTriviaParent)
+            if (EnableNullableAnalysis)
             {
-                if (current is StatementSyntax)
-                {
-                    return current;
-                }
+                // PROTOTYPE(nullable-api): This doesn't work for speculative models: the root is just the statement
+                // or expression being speculated on.
+                return Root;
```

Comment:
>return Root; [](start = 16, length = 12)

Can this affect error recovery binding for lambdas? For example, ```GetBoundLambdaOrQuery``` might not expect that we will jump out of an enclosing lambda. Also, it looks like this will "undo" laziness of IOperation tree creation, see ```GetBindingRootOrInitializer````. It feels like a better approach would be to not change the node returned by this function, but instead explicitly bind the root and analyze it when it is appropriate.

Reply:
I don't want to lose the debug assert above in nullable-enable code, so I'll add a default parameter to `GetBindingRoot` that will control this condition and only the appropriate places will set it to true.

---
In reply to: [269798837](https://github.com/dotnet/roslyn/pull/34429#discussion_r269798837) [](ancestors = 269798837)